### DISCUSSION TITLE
Easy Workflow to Create Custom Blocks

### DISCUSSION THREAD

**Discussion by afmarchetti** [WordPress/gutenberg#45746] [Developer Experience]
All the WordPress dev want to create block with php, js, css whitout npm and build worflow.
In a small business web site we haven't the time and budget to create a block with that system (local build) to make small changes, we need something quick and easy like this:
https://blockstudio.dev/

If you make something like this all the community will love Gutenberg and WordPress again.

This could change everything and make full site editing grow fast, think about that.

I went to WordCamp Italy, yesterday and all the devs agree with that, cheers keep the good work :)

**Comment by bph** [comment]
The tool you might want to check out is the Create Block scaffolding script built by the Gutenberg team. The recent course on [Introduction to Block Development](https://learn.wordpress.org/course/introduction-to-block-development-build-your-first-custom-block/) gives a great use case for it. 

You can also follow the [block editor handbook tutorial](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-create-block/), that has helped me to get started. 
 
There are other plugins and tools that give developers shortcuts to Custom Blocks for the transition. If it's ok with you @afmarchetti I convert this issue to a discussion so others can share their journey on Custom Block development as well. Issues are more for specific feature requests or bug fixes.

**Reply by afmarchetti** [reply]
I had follow the handbook but the process to create a block is too complex.

The main problem is that we need a local eviroment to make tiny change on the custom blocks, in small budget project this too time consuming and not sustainable.

Tools like block studio are exactly the type of experience we dev want, a simple folder with 3/4 files, no build...

**Reply by bph** [reply]
What kind of changes are you looking for?

**Reply by afmarchetti** [reply]
it will be awesome declare a custom block in a json file like this:

![Schermata 2022-11-14 alle 13 59 42](https://user-images.githubusercontent.com/8080212/201666494-ae15aa1c-9e3e-46d7-9f63-466a0689cee9.jpg)

and create a template in a php file like this:

![Schermata 2022-11-14 alle 13 57 54](https://user-images.githubusercontent.com/8080212/201666243-c2148294-0647-4ed7-befc-1b5c1a4b2176.jpg)

And style it in a css file like this:

![Schermata 2022-11-14 alle 13 57 58](https://user-images.githubusercontent.com/8080212/201666278-550a654b-89ba-45e6-9295-a2a125ffaf9f.jpg)
